dom/events/scrolling/scrollend-event-for-scroll-combination.html

```diff
@@ -0,0 +1,97 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <script src="/resources/testdriver.js"></script>
+    <script src="/resources/testdriver-actions.js"></script>
+    <script src="/resources/testdriver-vendor.js"></script>
+    <script src="scroll_support.js"></script>
+    <style>
+        #targetDiv {
+            width: 200px;
+            height: 200px;
+            overflow: scroll;
+        }
+
+        #innerDiv {
+            width: 400px;
+            height: 2000px;
+        }
+    </style>
+</head>
+<body style="margin:0" onload=runTest()>
+    <div id="targetDiv">
+        <div id="innerDiv">
+        </div>
+    </div>
+</body>
+
+<script>
+var target_div = document.getElementById('targetDiv');
+var user_scrollend_arrived = false;
+var programmatic_scrollend_arrived = false;
+var programmatic_scroll_done = false;
+var programmatic_scroll_top = 500;
+
+function onScroll(event) {
+    if (programmatic_scroll_done)
+        return;
+
+    // Perform programmatic scroll during user scrolling.
+    if (target_div.scrollTop > 50) {
+        target_div.scrollTop = programmatic_scroll_top;
+        programmatic_scroll_done = true;
```

Comment:
This is an interesting situation that is not clear in the spec the way it is written now. Particularly I'm not sure whether sending just one scrollend after all scrollings are done is better or sending two as you have here.

Reply:
Yes, make sense. Maybe we can discuss it on the issue? https://github.com/WICG/overscroll-scrollend-events/issues/4